mono/mini/interp/interp.c

```diff
@@ -2819,17 +2817,22 @@ interp_create_method_pointer_llvmonly (MonoMethod *method, gboolean unbox, MonoE
 	 * to use a ftndesc. The caller uses a normal signature, while the
 	 * entry functions use a gsharedvt_in signature, so wrap the entry function in
 	 * a gsharedvt_in_sig wrapper.
+	 * We use a gsharedvt_in_sig wrapper instead of an interp_in wrapper, because they
+	 * are mostly the same, and they are already generated. The exception is the
+	 * wrappers for methods with more than 8 arguments, those are different.
 	 */
-	wrapper = mini_get_gsharedvt_in_sig_wrapper (sig);
+	if (sig->param_count > MAX_INTERP_ENTRY_ARGS)
+		wrapper = mini_get_interp_in_wrapper (sig);
+	else
+		wrapper = mini_get_gsharedvt_in_sig_wrapper (sig);
```

Comment:
Why are we doing this on llvm as opposed to normal mode, where we always use the interp_in wrappers. They seem to be doing the same thing. Should we just use the interp_in_wrapper here for the sake of consistency ? Or is this meant to be an optimization that reduces the code size ?

Reply:
We are already generating the gsharedvt_in wrappers, so we can reuse them for this as well.

Reply:
Rewrote it to make it a bit clearer.